src/devices_models/devices/hydro_generation.jl

```diff
@@ -116,22 +116,25 @@ function _get_time_series(psi_container::PSIContainer,
         bus_number = PSY.get_number(PSY.get_bus(device))
         name = PSY.get_name(device)
         tech = PSY.get_tech(device)
-        # pf = sin(acos(PSY.get_powerfactor(PSY.get_tech(device))))
-        active_power = use_forecast_data ? PSY.get_rating(tech) : PSY.get_activepower(device)
+        pf = PSY.get_reactivepower(device) / PSY.get_rating(tech)       
         if use_forecast_data
+            active_power = PSY.get_rating(tech)
+            reactive_power = PSY.get_rating(tech) * pf
             ts_vector = TS.values(PSY.get_data(PSY.get_forecast(PSY.Deterministic,
                                                                 device,
                                                                 initial_time,
                                                                 "get_rating")))
         else
+            active_power = PSY.get_activepower(device)
+            reactive_power = PSY.get_reactivepower(device) 
             ts_vector = ones(time_steps[end])
         end
         range_data = DeviceRange(name, get_constraint_values(device))
         _device_services!(range_data, device, model)
         push!(constraint_data, range_data)
         push!(active_timeseries, DeviceTimeSeries(name, bus_number, active_power, ts_vector,
                                                  range_data))
-        push!(reactive_timeseries, DeviceTimeSeries(name, bus_number, active_power,
+        push!(reactive_timeseries, DeviceTimeSeries(name, bus_number, reactive_power,
```

Comment:
@daniel-thom, just a heads up that if you aren't going to use the rest of these changes, you should probably pick up this one.